boardswarm-cli/src/ui.rs

```diff
@@ -203,17 +289,30 @@ pub async fn run_ui(
                 }
                 Some(input) = input_rx.recv() => {
                     match input {
-                        Input::Up => { terminal.scroll_up(); },
-                        Input::Down => { terminal.scroll_down(); },
-                        Input::ScrollReset => { terminal.scroll_reset(); },
-                        _ => (),
+                        InputMsg::Input(input) =>
+                            match input {
+                                Input::Up => { terminal.scroll_up(); },
+                                Input::Down => { terminal.scroll_down(); },
+                                Input::ScrollReset => { terminal.scroll_reset(); },
+                                _ => (),
+                            },
+                        InputMsg::Error(e) => { terminal.set_message(e); },
+                    }
+                }
+                Some(new_state) = state_rx.recv() => {
+                    let state: String = new_state;
+                    match state.as_str() {
+                        "on" => terminal.set_functioning("ON".to_string()),
+                        "off" => terminal.set_non_functioning("OFF".to_string()),
```

Comment:
these are devie modes not terminal modes though; functioning/non-functioning is weird jargon as welll ;  why not just on/off/other(string) ?

Reply:
I'm okay with renaming `TerminalMode` to `DeviceMode`.

Functioning/NonFunctioning: Perhaps there will be states which want to be displayed the same way "off" or "on" is displayed. Let's imagine there will appear a state called "relay" which wants to be displayed black on green. Then the match arm becomes:

```
"relay" => terminal.set_functioning("RELAY".to_string()),
```

Otherwise it will have to be:

```
"relay" => terminal.set_on("RELAY".to_string()),
```

Reply:
can you please not yell in strings ;) Yes there might be cases we may want to do differently (or maybe we could stick a hint in the protocol for broad categories), but regardless we can simply add more "known" items in the enum. 

Fwiw it would also be nicer to implement From<&str> for your mode enum